app/models/wpcc/contributions_calendar.rb

```diff
@@ -27,5 +22,16 @@ def periods
         user_input_employer_percent: employer_percent
       ).filter
     end
+
+    def percents_for(period)
+      Wpcc::PeriodContributionCalculator.new(
+        name: period.name,
+        employee_percent: period.employee_percent,
+        employer_percent: period.employer_percent,
+        eligible_salary: eligible_salary,
+        salary_frequency: salary_frequency,
+        tax_relief_percent: period.tax_relief_percent
+      )
+    end
```

Comment:
IMHO, I don't thing we need to move this to a private method. There is no logic here, only instantiation and calls that will not be reused at all and is a one line (it is only multiple due to the line length limitation). I will be honest that usually I "scratch my head" every time that I saw a class with a lot of private methods (it is not this case but I want to avoid the broken window) because every time moving logic to a private methods means that a public interface it is requiring to do a *heavy* work which is delegated to a bunch of private methods which sounds a smell to me. Anyway, it is not a big blocker too.